mxcubecore/BaseHardwareObjects.py

```
@@ -163,15 +168,7 @@ def objects_by_role(self) -> Dict[str, Union[Self, None]]:
         Returns:
             OrderedDict[str, Union[Self, None]]: Contained objects mapped by role.
         """
-        result = {}
-        for tag in self._roles:
```

I had another lesson in the depths of python ;). This is also a quite nasty one. Any `Exception` raised in a property is translated/changed into an `AttributeError` and the original exception gets hidden. This gets especially confusing when one have been debugging `getattr` and `setattr` for some time.

I guess one should never explicitly raise anything from a property ?